integration-tests/smoke/ccip/ccip_token_price_updates_test.go

```diff
@@ -0,0 +1,141 @@
+package smoke
+
+import (
+	"math"
+	"math/big"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"golang.org/x/exp/maps"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/config"
+	"github.com/smartcontractkit/chainlink-common/pkg/utils/tests"
+	"github.com/smartcontractkit/chainlink-protos/job-distributor/v1/node"
+	"github.com/smartcontractkit/chainlink/deployment"
+	"github.com/smartcontractkit/chainlink/deployment/ccip/changeset"
+	testsetups "github.com/smartcontractkit/chainlink/integration-tests/testsetups/ccip"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/ccip/generated/fee_quoter"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+)
+
+func Test_CCIPTokenPriceUpdates(t *testing.T) {
+	lggr := logger.TestLogger(t)
+	ctx := changeset.Context(t)
+
+	var tokenPriceExpiry = 5 * time.Second
+	e, _, _ := testsetups.NewLocalDevEnvironmentWithDefaultPrice(t, lggr, &changeset.TestConfigs{
+		OCRConfigOverride: func(params changeset.CCIPOCRParams) changeset.CCIPOCRParams {
+			params.CommitOffChainConfig.TokenPriceBatchWriteFrequency = *config.MustNewDuration(tokenPriceExpiry)
+			return params
+		},
+	})
+	state, err := changeset.LoadOnchainState(e.Env)
+	require.NoError(t, err)
+	require.NoError(t, changeset.AddLanesForAll(e.Env, state))
+
+	allChainSelectors := maps.Keys(e.Env.Chains)
+	assert.GreaterOrEqual(t, len(allChainSelectors), 2, "test requires at least 2 chains")
+
+	sourceChain1 := allChainSelectors[0]
+	//sourceChain2 := allChainSelectors[1]
+
+	feeQuoter1 := state.Chains[sourceChain1].FeeQuoter
+
+	feeTokensChain1, err := feeQuoter1.GetFeeTokens(&bind.CallOpts{Context: ctx})
+	require.NoError(t, err)
+	t.Logf("feeTokens: %v", feeTokensChain1)
+
+	tokenPricesBefore, err := feeQuoter1.GetTokenPrices(&bind.CallOpts{Context: ctx}, feeTokensChain1)
+	require.NoError(t, err)
+	t.Logf("tokenPrices: %v", tokenPricesBefore)
+
+	// assert token prices updated due to time expiration
+	assert.Eventually(t, func() bool {
+		tokenPricesNow, err := feeQuoter1.GetTokenPrices(&bind.CallOpts{Context: ctx}, feeTokensChain1)
+		require.NoError(t, err)
+		t.Logf("tokenPrices: %v", tokenPricesNow)
+
+		// both tokens should have same price but different timestamp since there was an update due to time deviation
+		for i, price := range tokenPricesNow {
+			if tokenPricesBefore[i].Timestamp == price.Timestamp {
+				tokenPricesBefore = tokenPricesNow
+				return false // timestamp is the same
+			}
+			if tokenPricesBefore[i].Value.Cmp(price.Value) != 0 {
+				tokenPricesBefore = tokenPricesNow
+				return false // price was updated
+			}
+		}
+		t.Log("time expiration assertions complete")
+		return true
+	}, tests.WaitTimeout(t), 500*time.Millisecond)
+
+	// disable oracles to prevent price updates while we manually edit token prices
+	var disabledNodes []string
```

Comment:
disabling oracles could be extracted to a separate function, it would make the test easier to read